src/state_variables.jl

```diff
@@ -72,12 +98,15 @@ function StateVariables(
     )
 end
 
-# TODO: just take the first prognostic variable for the eltype, bad idea?
-Base.eltype(vars::StateVariables) = eltype(vars.prognostic[1])
```

Comment:
@maximilian-gelbrecht Turns out this was in fact a bad idea. There can be model configurations (especially in tests) that do not have prognostic variables. Sorry, should have caught that before.

But I went ahead and re-added the `NF` type argument to `StateVariables`.